src/util/mod.rs

```diff
@@ -156,21 +156,33 @@ where
 
 /// Check for an expected error.
 #[cfg(test)]
-pub fn expect_err<T: core::fmt::Debug, E: core::fmt::Debug>(result: Result<T, E>, err_msg: &str) {
+pub fn expect_err<T: core::fmt::Debug, E: core::fmt::Debug + core::fmt::Display>(
+    result: Result<T, E>,
+    err_msg: &str,
+) {
     use alloc::format;
-    assert!(
-        result.is_err(),
-        "expected error containing '{}', got success {:?}",
-        err_msg,
-        result
-    );
-    let err = result.err();
-    assert!(
-        format!("{:?}", err).contains(err_msg),
-        "unexpected error {:?}, doesn't contain '{}'",
-        err,
-        err_msg
-    );
+    match result {
+        Ok(_) => {
+            panic!(
```

Comment:
This is failing a CI check because we've got a local convention that things that `panic!` need a comment of the form "safe: ..." on the same line to document why it's OK to panic.  (Unfortunately the check is very primitive, so it doesn't notice that this is `#[cfg(test)]` code.)

Reply:
Nope, it's an even dumber check than that :-(  (`./scripts/check_format.sh`). 

Let me see if I can make it less pesky&hellip;

Reply:
Oh, interesting challenge. If the `panic!` spreads across multiple lines, `cargo fmt` wants to put the comment on a line of its own.

Might be simpler for me to rewrite this as an `assert!(false...)`.

Reply:
Yeah, `cargo fmt` and `check_format.sh` fight each other.   

Shifting to `assert` should work though &hellip; the check is just for `panic` and the red things on <https://tinyurl.com/rust-transform>.